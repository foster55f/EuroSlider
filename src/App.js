import React from 'react';
import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div>Hello Soc</div>
  );
}

export default App;
