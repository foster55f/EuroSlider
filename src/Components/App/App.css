.App {
  background-color: #bdbdbdad;
  height: 100vh;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.game-card{
  color: black;
  background-color: black;
  margin: 12px;
}

button {
  padding: .5vw;
  font-size: 1.3vw;
  background-color: white;
  border: black;
  margin-left: 20%;
  left:65%;
  border-radius: 5px;
}


.title-font {
  color: aliceblue;
  font-size: 23px;
}

.header-favorite{
  margin-left: 60%;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.header-background {
    background-color:black;
    height: 20vh;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

nav {
    background-color: black;
    display: flex;
    align-items: center;
    padding: 1vw;
    height: 50px;
    margin-top: -40px;
    color: white
}

.header-title {
  color:white;
  font-size: 4vw;
  margin-left: 2%;
}

.App-link {
  color: #61dafb;
}

.game-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  margin-left: auto;
  margin-right: auto;
  width: 90%;
  text-align: center;
}
.game-img {
  background-size: cover;
  height: 50vh;
  background-repeat: no-repeat;
  width: 25vw;
  margin-left: auto;
  margin-right: auto;
  background-position: center;
}

.share-button{
  font-size: smaller;
  text-align: center;
  margin: auto;
  margin-top: 5px;
  margin-bottom: 5px;
  background-color: dodgerblue;
  outline: none;
}

.search-input {
  width: 30%;
  height: 30px;
  border-radius: 10px;
  margin-top:10px;
  font-size: 20px;
  text-align: center;
  margin-left:10px;
}

.favorite-button{
  background-color: dodgerblue;
  text-align: center;
  margin: auto;
  outline: none;
}

.userName{
  margin:auto;
  width: 50%;
  height: 30px;
  border-radius: 10px;
  font-size: 20px;
  text-align: center
}

.password{
    margin:auto;
    width: 30%;
    height: 30px;
    border-radius: 10px;
    font-size: 20px;
    text-align: center
}

.searchContainer{
  display: flex;
  flex-direction: row;

}

.searchButton {
  text-align: center;
  margin-top: 10px;
  background-color:dodgerblue;
}

.user-entry {
  display: flex;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
