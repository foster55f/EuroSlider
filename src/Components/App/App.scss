@import '../../variables.scss';

.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}
.game-img {
  background-size: cover;
  height: 50vh;
  background-repeat: no-repeat;
  width: 25vw;
  margin-left: auto;
  margin-right: auto;
  background-position: center;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
