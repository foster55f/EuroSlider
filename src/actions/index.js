export const addGames = games => ({
    type: 'ADD_GAMES',
    games
})

export const filterGames = displayGames => ({
    type: 'FILTER_GAMES',
    displayGames
})

