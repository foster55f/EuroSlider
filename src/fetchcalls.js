export const retrieveGames = () => {
    return fetch('https://www.scorebat.com/video-api/v1/')
    .then(response => {
        if (!response.ok) {
          throw Error('Error fetching games');
        }
        return response.json();
      })
}