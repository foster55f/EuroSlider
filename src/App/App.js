import React from 'react';
import './App.css';

class App extends React.Component {
  constructor() {
    super();
  }

  componentDidMount() {
    fetch('https://www.scorebat.com/video-api/v1/')
      .then(response => response.json())
      .then(data => console.log(data))
  }

  render() {
    return (
      <></>
    )
  }
}

export default App;
