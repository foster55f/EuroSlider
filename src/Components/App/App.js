import './App.css';
import HighlightContainer from '../HighlightContainer/HighlightContainer';
import SearchHighlightContainer from '../SearchHighlightContainer/SearchHighlightContainer';
import FavoriteContainer from '../FavoriteContainer/FavoriteContainer';
import { addGames } from '../../actions';
import Login from '../Login/Login';
import SearchForm from '../SearchForm/SearchForm';
import { Route } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { retrieveGames } from '../../fetchcalls';
import { useDispatch} from 'react-redux';
import React, { useEffect } from 'react';



const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    retrieveGames('https://www.scorebat.com/video-api/v1/')
      .then(games => {
        dispatch(addGames(games))
    })
  }, []);

  const filterSearch = (search) => {
    this.props.history.push(`/yoursearch`)
    this.props.filterGames(search)
  }

  const goToHomePage = () => {
    this.props.history.push(`/`)
}

    return (
      <div className="App">
        <Route
          exact path="/"
          render={() => {
            return (
              <>
              <Login />
                {/* <SearchForm search={this.filterSearch} /> */}
              <HighlightContainer />                
              </>
            )
          }}
        />
        <Route
          exact path="/yoursearch"
          render={() => {
            return (
              <>
              <Login />  
                <Link to='/' >
                  Go Back Home
                </Link>  
                <SearchForm search={this.filterSearch} />                 
                <SearchHighlightContainer />
              </>
            )
          }}
        />
        <Route
          exact path="/yourfavorites"
          render={() => {
            return (
              <>    
                <Login />    
                  <Link to='/' >
                  Go Back Home
                  </Link>   
                <FavoriteContainer />               
              </>              
            )
          }}
        />
      </div>
    )
  }

export default App

