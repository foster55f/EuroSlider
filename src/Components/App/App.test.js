import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { retrieveGames } from '../../fetchcalls';
import { addGames } from '../../actions';

jest.mock('../../fetchcalls.js')


describe('App', () => {
  let wrapper;
  beforeEach(() => {
    retrieveGames.mockImplementation(() => {
        return Promise.resolve([{
          title: "Smouha - Wadi Degla",
          embed: "url"
      },
      {
          title: "foster - barcelona",
          embed: "url"
      },
      ]);
    });
});
  it('should match the App snapshot', () => {
    wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should invoke retrieveGames after componentDidMount triggers', () => {
    wrapper = shallow(<App />);
    expect(retrieveGames).toHaveBeenCalled();
  });

  describe('mapsStateToProps', () => {
    it('should return only movies and loading properties from the store', () => {
        const mockState = {
            games: [],
            displayGames: [],
        };
        const expected = {
            games: [],
            displayGames: []
        };
        const mappedProps = mapStateToProps(mockState);

        expect(mappedProps).toEqual(expected);
    });
});

describe('mapDispatchToProps', () => {
    it('calls dispatch with an addMovies action when addMovies is called',
        () => {
            const mockDispatch = jest.fn();
            const actionToDispatch = addGames([{ sample: 'game' }]);
            const mappedProps = mapDispatchToProps(mockDispatch);

            mappedProps.addGames([{ sample: 'game' }]);

            expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });  
})
