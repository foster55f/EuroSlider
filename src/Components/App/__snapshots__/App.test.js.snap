// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the App snapshot 1`] = `
<div
  className="App"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/yoursearch"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/user/favorites"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/user"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/yourfavorites"
    render={[Function]}
  />
</div>
`;
